`TestDeleteDocuments` flakiness

## Failure

### Relevant snippet

```
WARNING: DATA RACE
Read at 0x00c000371410 by goroutine 791:
  github.com/gravitational/teleport/lib/backend/firestore.(*mockFirestoreServer).BatchWrite()
      /Users/mike/go/src/github.com/gravitational/teleport/lib/backend/firestore/firestorebk_test.go:205 +0xe0
  cloud.google.com/go/firestore/apiv1/firestorepb._Firestore_BatchWrite_Handler()
      /Users/mike/go/pkg/mod/cloud.google.com/go/firestore@v1.9.0/apiv1/firestorepb/firestore.pb.go:5176 +0x1f4
  google.golang.org/grpc.(*Server).processUnaryRPC()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1336 +0x1038
  google.golang.org/grpc.(*Server).handleStream()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1704 +0xbec
  google.golang.org/grpc.(*Server).serveStreams.func1.2()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:965 +0xa4

Previous write at 0x00c000371410 by goroutine 789:
  github.com/gravitational/teleport/lib/backend/firestore.(*mockFirestoreServer).BatchWrite()
      /Users/mike/go/src/github.com/gravitational/teleport/lib/backend/firestore/firestorebk_test.go:205 +0x188
  cloud.google.com/go/firestore/apiv1/firestorepb._Firestore_BatchWrite_Handler()
      /Users/mike/go/pkg/mod/cloud.google.com/go/firestore@v1.9.0/apiv1/firestorepb/firestore.pb.go:5176 +0x1f4
  google.golang.org/grpc.(*Server).processUnaryRPC()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1336 +0x1038
  google.golang.org/grpc.(*Server).handleStream()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:1704 +0xbec
  google.golang.org/grpc.(*Server).serveStreams.func1.2()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:965 +0xa4

Goroutine 791 (running) created at:
  google.golang.org/grpc.(*Server).serveStreams.func1()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:963 +0x428
  google.golang.org/grpc/internal/transport.(*http2Server).operateHeaders()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/internal/transport/http2_server.go:620 +0x2880
  google.golang.org/grpc/internal/transport.(*http2Server).HandleStreams()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/internal/transport/http2_server.go:662 +0x2cc
  google.golang.org/grpc.(*Server).serveStreams()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:949 +0x24c
  google.golang.org/grpc.(*Server).handleRawConn.func1()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:891 +0x54

Goroutine 789 (running) created at:
  google.golang.org/grpc.(*Server).serveStreams.func1()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:963 +0x428
  google.golang.org/grpc/internal/transport.(*http2Server).operateHeaders()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/internal/transport/http2_server.go:620 +0x2880
  google.golang.org/grpc/internal/transport.(*http2Server).HandleStreams()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/internal/transport/http2_server.go:662 +0x2cc
  google.golang.org/grpc.(*Server).serveStreams()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:949 +0x24c
  google.golang.org/grpc.(*Server).handleRawConn.func1()
      /Users/mike/go/pkg/mod/google.golang.org/grpc@v1.53.0/server.go:891 +0x54
==================
--- FAIL: TestDeleteDocuments (0.00s)
    --- FAIL: TestDeleteDocuments/commit_success (0.19s)
        testing.go:1319: race detected during execution of test
FAIL
FAIL	github.com/gravitational/teleport/lib/backend/firestore	2.221s
FAIL
```

--- a/lib/backend/firestore/firestorebk_test.go
+++ b/lib/backend/firestore/firestorebk_test.go
@@ -22,6 +22,7 @@ import (
 	"os"
 	"reflect"
 	"strings"
+	"sync"
 	"testing"
 	"time"
 	"unsafe"
@@ -190,6 +191,7 @@ type mockFirestoreServer struct {
 	// in the future.
 	firestorepb.FirestoreServer
 
+	mu   sync.RWMutex
 	reqs []proto.Message
 
 	// If set, Commit returns this error.
@@ -202,7 +204,10 @@ func (s *mockFirestoreServer) BatchWrite(ctx context.Context, req *firestorepb.B
 		return nil, fmt.Errorf("x-goog-api-client = %v, expected gl-go key", xg)
 	}
 
+	s.mu.Lock()
 	s.reqs = append(s.reqs, req)
+	s.mu.Unlock()
+
 	if s.commitErr != nil {
 		return nil, s.commitErr
 	}
@@ -312,12 +317,14 @@ func TestDeleteDocuments(t *testing.T) {
 			}
 
 			var committed int
+			mockFirestore.mu.RLock()
 			for _, req := range mockFirestore.reqs {
 				switch r := req.(type) {
 				case *firestorepb.BatchWriteRequest:
 					committed += len(r.Writes)
 				}
 			}
+			mockFirestore.mu.RUnlock()
 
 			require.Equal(t, tt.documents, committed)
 

